src/session/sidebar/row.rs

```diff
@@ -72,223 +73,255 @@ mod imp {
+                    move |args| -> String {
+                        let date = args[1].get::<i32>().unwrap();
+
+                        let datetime_now = glib::DateTime::new_now_local().unwrap();
+                        let datetime = glib::DateTime::from_unix_utc(date as i64)
+                            .and_then(|t| t.to_local())
+                            .unwrap();
+
+                        let hours_difference = datetime_now.difference(&datetime) / 3600000000;
+                        let days_difference = hours_difference / 24;
+
+                        if hours_difference <= 16 {
+                            // Show the time
+                            let mut time = datetime.format("%X").unwrap().to_string();
+
+                            // Remove seconds
+                            time.replace_range(5..8, "");
+                            time
+                        } else if days_difference < 6 {
+                            // Show the day of the week
+                            datetime.format("%a").unwrap().to_string()
+                        } else if days_difference < 364 {
+                            // Show the day and the month
+                            datetime.format("%d %b").unwrap().to_string()
                         } else {
-                            "group-chats-notification-settings"
+                            // Show the entire date
+                            datetime.format("%x").unwrap().to_string()
                         }
-                    }
-                },
-            );
-
-            let unread_count_label_css_expression = gtk::ClosureExpression::new(
-                |args| {
-                    let notification_settings =
-                        args[1].get::<BoxedChatNotificationSettings>().unwrap().0;
-
-                    let scope_notification_settings =
-                        args[2].get::<BoxedScopeNotificationSettings>().unwrap().0;
-
-                    vec![
-                        "unread-count".to_string(),
-                        if notification_settings.use_default_mute_for {
-                            if scope_notification_settings
-                                .map(|s| s.mute_for > 0)
-                                .unwrap_or(notification_settings.mute_for > 0)
-                            {
+                    },
+                    &[date_expression.upcast()],
+                );
+                timestamp_expression.bind(&*self_.timestamp_label, "label", Some(chat));
+
+                // Last message label bindings
+                let content_expression = gtk::PropertyExpression::new(
+                    Message::static_type(),
+                    Some(&last_message_expression),
+                    "content",
+                );
+                // FIXME: the sender name should be part of this expression.
+                let stringified_message_expression = gtk::ClosureExpression::new(
+                    move |args| {
```

Comment:
Not part of the PR, but as we do massive refactorings anyway: This `move` is not needed